### DISCUSSION TITLE
Ability to provide own encoding

### DISCUSSION THREAD

**Discussion by Xottab-DUTY** [foonathan/lexy#192] [Q&A] [answered]
Currently lexy has static assertions that only allow predefined encodings.
I want to create my custom `word_encoding` so I can read 2 bytes instead of just byte one by one.

**Comment by Xottab-DUTY** [comment]
Well, now I'm unsure if this exact feature is needed, maybe there's a better solution for my use case?

Pseudo-code:
```cpp
struct production
{
    static constexpr auto rule = []
    {
        return lexy::bytes<2>.range(65534, 65535);
    }
}
```

I want to read 2 bytes and validate that they are in the allowed range. Yes, it's possible to validate each byte, but this is unwanted :)

**Reply by foonathan** [reply]
What's the problem with parsing a `dsl::buint16` and validating it later, e.g. when constructing the data structure. Generally, lexy is only concerned with syntactic parts of the input and semantic requirements are best verified in a later step where you have more flexibility.

**Comment by foonathan** [comment, marked as answer]
Custom encodings aren't supported, and I don't plan on supporting them. An input that consists of 2 byte words is still a byte input, you just want rules that read two bytes at a time.
